src/gravel/controllers/nodes/systemdisk.py

```diff
@@ -55,30 +55,64 @@ class OverlayError(GravelError):
 class MountEntry(BaseModel):
     source: str
     dest: str
+    overlay: bool
 
 
 AQR_SYSTEM_PATH = "/var/lib/aquarium-system"
 
 
 def get_mounts() -> List[MountEntry]:
+    """Obtain mounts from the system."""
+
+    def _get_overlay_src(opts: str) -> str:
+        optlst: List[str] = opts.split(",")
+        upper: Optional[str] = None
+        for e in optlst:
+            if e.startswith("upperdir"):
+                path = Path(e.split("=")[1])
+                upper = path.parent.as_posix()
+        if upper is None or len(upper) == 0:
+            raise OverlayError(
+                f"Unable to find upper dir from options: {opts}."
+            )
+        return upper
+
     proc: Path = Path("/proc/mounts")
     assert proc.exists()
 
     lst: List[MountEntry] = []
     with proc.open(mode="r", encoding="utf-8") as f:
         for line in f.readlines():
             fields: List[str] = line.split(" ")
-            if len(fields) < 2:
+            if len(fields) < 4:
                 continue
-            src = fields[0].strip()
-            dst = fields[1].strip()
-            lst.append(MountEntry(source=src, dest=dst))
+
+            fields = [x.strip() for x in fields]
+            src, dst, fstype, opts = fields[0:4]
+            is_overlay = fstype == "overlay"
+            if is_overlay:
+                try:
+                    src = _get_overlay_src(opts)
+                except OverlayError as e:
+                    raise OverlayError(
+                        f"Unable to find overlay source for {dst}: {e.message}"
+                    )
+
+            lst.append(MountEntry(source=src, dest=dst, overlay=is_overlay))
     return lst
 
 
+async def lvm(args: str) -> str:
```

Comment:
Isn't it better to use a `List[str]` here instead of using `shlex.split`? I think the developer better knows how to split the commands, especially when it comes to more complicated command args.